aurora_dsql_django/base.py

```diff
@@ -126,16 +127,20 @@ def uuid_get_prep_value(self, value):
             return Field.get_prep_value(self, value)
 
         def uuid_to_python(self, value):
-            """Convert UUID strings to UUID objects, handle both UUIDs and integers gracefully."""
-            if value is None:
+            """Convert provided value to a UUID where possible."""
+            if value is None or isinstance(value, uuid.UUID):
                 return value
             if isinstance(value, str):
                 try:
                     return uuid.UUID(value)
                 except ValueError:
-                    # If it's not a valid UUID, treat it as the original field would.
-                    return Field.to_python(self, value)
-            return value
+                    pass
+
+            raise ValidationError(
+                self.error_messages["invalid"],
```

Comment:
Please improve the error message to let the customer know what is actually invalid. Something in the lines of `invalid uuid '<the-invalid-value-here>'` would help.

Reply:
I think these are defined in a standard map of error messages, I don't think we want this to be custom as Django seems to use them to determine what to show in the UI.

This error code snippet is directly copied from the `AutoField` implementation that uses integers.

Reply:
> I think it will show a more helpful message to the user.

Great! Thanks for the context. Just make sure the invalid value is shown before the code is merged.